modules/core/src/lib/deck.ts

```diff
@@ -1021,28 +1020,28 @@ export default class Deck {
       clearCanvas?: boolean;
     }
   ) {
-    const {device, gl} = this.layerManager?.context;
+    const {device, gl} = this.layerManager!.context;
 
     setGLParameters(device, this.props.parameters);
 
     this.props.onBeforeRender({device, gl});
 
     const opts = {
       target: this.props._framebuffer,
-      layers: this.layerManager?.getLayers(),
-      viewports: this.viewManager?.getViewports(),
-      onViewportActive: this.layerManager?.activateViewport,
-      views: this.viewManager?.getViews(),
+      layers: this.layerManager!.getLayers(),
```

Comment:
I guess an alternative is to ignore the error at the declaration site, instead of sprinkling exclamation marks through the code. An advantage being that we have less places in the code where we explicitly disable type checks.